src/components/vm/vmReplaceSpiceDialog.jsx

```diff
@@ -30,17 +36,28 @@ import { fmt_to_fragments } from 'utils.jsx';
 import { NeedsShutdownAlert } from '../common/needsShutdown.jsx';
 import { domainReplaceSpice } from '../../libvirtApi/domain.js';
 
+import './vmReplaceSpiceDialog.scss';
+
 const _ = cockpit.gettext;
 
-export const ReplaceSpiceDialog = ({ vm }) => {
+export const ReplaceSpiceDialog = ({ vm, vms }) => {
+    const spiceVMs = vms?.filter(vm => vm.inactiveXML?.hasSpice);
+    // sort spicVMs by name, put the current VM first
+    spiceVMs?.sort((a, b) => (a.name === vm.name) ? -1 : (b.name === vm.name) ? 1 : a.name.localeCompare(b.name));
+    const isMultiple = spiceVMs?.length > 1;
+
     const Dialogs = useDialogs();
     const [error, dialogErrorSet] = useState(null);
     const [inProgress, setInProgress] = useState(false);
+    const [selected, setSelected] = useState(isMultiple ? [spiceVMs[0].id] : []);
```

Comment:
It's quite confusing how the "primary" VM (passed in `vm`) is handled re the various lists of vms. The primary is part of `spiceVms`, and might be part of `selected` (but only if `isMultiple` is true), so it might also be part of `selectedVMs` (in the multi case), and wouldn't it then be converted twice?

Reply:
Right, the dialog needs to handle both cases (single and multi-VM) separately. Thanks for spotting!